src/easylink/pipeline_schema_constants.py

```diff
@@ -0,0 +1,172 @@
+                    InputSlot(
+                        "step_1b_main_input",
+                        "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                        validate_input_file_dummy,
+                    )
+                ],
+                output_slots=["step_1b_main_output"],
+            ),
+        ],
+        edges=[
+            Edge("step_1a", "step_1b", "step_1a_main_output", "step_1b_main_input"),
+        ],
+        slot_mappings={
+            "input": [("step_1a", "step_1_main_input", "step_1a_main_input")],
+            "output": [("step_1b", "step_1_main_output", "step_1b_main_output")],
+        },
+    ),
+    ImplementedStep(
+        "step_2",
+        input_slots=[
+            InputSlot(
+                "step_2_main_input",
+                "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                validate_input_file_dummy,
+            )
+        ],
+        output_slots=["step_2_main_output"],
+    ),
+    ImplementedStep(
+        "step_3",
+        input_slots=[
+            InputSlot(
+                "step_3_main_input",
+                "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                validate_input_file_dummy,
+            )
+        ],
+        output_slots=["step_3_main_output"],
+    ),
+    ImplementedStep(
+        "step_4",
+        input_slots=[
+            InputSlot(
+                "step_4_main_input",
+                "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                validate_input_file_dummy,
+            ),
+            InputSlot(
+                "step_4_secondary_input",
+                "DUMMY_CONTAINER_SECONDARY_INPUT_FILE_PATHS",
+                validate_input_file_dummy,
+            ),
+        ],
+        output_slots=["step_4_main_output"],
+    ),
+    ResultStep(
+        "results_schema",
+        input_slots=[InputSlot("result", None, validate_input_file_dummy)],
+        output_slots=[],
+    ),
+]
+SCHEMA_EDGES = [
+    Edge(
+        in_node="input_data_schema",
+        out_node="step_1",
+        output_slot="file1",
+        input_slot="step_1_main_input",
+    ),
+    Edge(
+        in_node="input_data_schema",
+        out_node="step_4",
+        output_slot="file1",
+        input_slot="step_4_secondary_input",
+    ),
+    Edge(
+        in_node="step_1",
+        out_node="step_2",
+        output_slot="step_1_main_output",
+        input_slot="step_2_main_input",
+    ),
+    Edge(
+        in_node="step_2",
+        out_node="step_3",
+        output_slot="step_2_main_output",
+        input_slot="step_3_main_input",
+    ),
+    Edge(
+        in_node="step_3",
+        out_node="step_4",
+        output_slot="step_3_main_output",
+        input_slot="step_4_main_input",
+    ),
+    Edge(
+        in_node="step_4",
+        out_node="results_schema",
+        output_slot="step_4_main_output",
+        input_slot="result",
+    ),
+]
+ALLOWED_SCHEMA_PARAMS = {"development": (SCHEMA_NODES, SCHEMA_EDGES)}
+
+TESTING_NODES = [
```

Comment:
What are these TESTING_* things? Should they live here?

Reply:
I think we talked about splitting the different schemas into their own separate files, which probably makes sense to do here.